onnxruntime/core/providers/cuda/math/softmax.h

```diff
@@ -36,12 +36,14 @@ class Softmax final : public CudaKernel {
  public:
   Softmax(const OpKernelInfo& info) : CudaKernel{info} {
     info.GetAttrOrDefault("axis", &axis_, static_cast<int64_t>(1));
+    log_softmax_ = info.GetKernelDef().OpName() == "LogSoftmax";
   }
 
   Status ComputeInternal(OpKernelContext* context) const override;
 
  private:
   int64_t axis_;
+  bool log_softmax_;
```

Comment:
>log_softmax_ [](start = 7, length = 12)

This would work, but it's still better to have it as template parameter. 

see /onnxruntime/core/providers/cuda/nn/dropout.h
Dropout Kernel is also used for TrainableDropout op

Reply:
Got it. Will make note how it's done for _dropout_ and will defer "is_log_softmax as template parameter change" to next PR re: softmax.

Reply:
:thumbsup: 

---
In reply to: [466583790](https://github.com/microsoft/onnxruntime/pull/4726#discussion_r466583790) [](ancestors = 466583790)